.github/workflows/update-branches.yaml

```diff
@@ -0,0 +1,51 @@
+name: Auto-update branches
+
+on:
+  push:
+    branches:
+      - main
+      - 'release-[0-9]+.[0-9]+'
+
+permissions:
+  contents: read
+
+jobs:
+  update:
+    name: "${{ matrix.patch }}"
+    permissions:
+      contents: write  # for Git to git push
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        patch: ["strict", "moonray"]
```

Comment:
Could you please add the patch matrix to our nightly run as well or add a follow-up card?